sources/packages/backend/apps/api/src/route-controllers/student/_tests_/e2e/student.institutions.controller.searchStudents.e2e-spec.ts

```diff
@@ -0,0 +1,470 @@
+    });
+
+    await saveFakeApplicationDisbursements(
+      appDataSource,
+      {
+        institution: collegeF,
+        institutionLocation: collegeFLocation,
+        student,
+      },
+      {
+        applicationStatus: ApplicationStatus.Submitted,
+      },
+    );
+    const endpoint = "/institutions/student/search";
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFUser,
+    );
+    const searchPayload = {
+      appNumber: "",
+      firstName: student.user.firstName,
+      lastName: "",
+      sin: "",
+    };
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .send(searchPayload)
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.CREATED)
+      .then((response) => {
+        expect(response.body).toEqual(
+          expect.arrayContaining([
+            {
+              id: student.id,
+              firstName: student.user.firstName,
+              lastName: student.user.lastName,
+              birthDate: student.birthDate,
+              sin: student.sinValidation.sin,
+            },
+          ]),
+        );
+      });
+  });
+
+  it("Should find the student by sin when student has at least one application submitted for the institution.", async () => {
+    // Arrange
+    // Student who has application submitted to institution.
+    let student = await saveFakeStudent(appDataSource);
+    // Refresh student object to get a birthDate with date only
+    student = await studentRepo.findOne({
+      where: { id: student.id },
+      relations: { sinValidation: true },
+    });
+
+    await saveFakeApplicationDisbursements(
+      appDataSource,
+      {
+        institution: collegeF,
+        institutionLocation: collegeFLocation,
+        student,
+      },
+      {
+        applicationStatus: ApplicationStatus.Submitted,
+      },
+    );
+    const endpoint = "/institutions/student/search";
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFUser,
+    );
+    const searchPayload = {
+      appNumber: "",
+      firstName: "",
+      lastName: "",
+      sin: student.sinValidation.sin,
+    };
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .post(endpoint)
+      .send(searchPayload)
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.CREATED)
+      .expect([
+        {
+          id: student.id,
+          firstName: student.user.firstName,
+          lastName: student.user.lastName,
+          birthDate: student.birthDate,
+          sin: student.sinValidation.sin,
+        },
+      ]);
+  });
+
+  it("Should not find the student when application submitted and cancelled before the assessment.", async () => {
+    // Arrange
+    // Student who has application submitted to institution.
+    const student = await saveFakeStudent(appDataSource);
+
+    const application = await saveFakeApplicationDisbursements(
```

Comment:
Other `saveFakeApplicationSomething` helpers can be created. In this case, we are calling a method to create the assessments/disbursements/values and then removing them.
For this particular case, maybe the `createFakeApplication` would be enough.

Reply:
I create a `saveFakeApplication()` that can be called just for these cases when you need a application. I refactored `saveFakeApplicationDisbursements` to call it.